### PR TITLE
Do not allow negative router reload intervals

### PR DESCRIPTION
**Pull request by eparis**
Default to 5 seconds.

### CODE REVIEW

**Comment by eparis** [comment]
[test]

**Comment by eparis** [comment]
@smarterclayton @ramr @deads2k

**Comment by eparis** [comment]
The bug report is about being able to set the value negative in the DC. I don't think we can prevent that, but we can at least warn about it at run time.

**Comment by eparis** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1311459

File: pkg/router/template/router.go [hunk 1]
```diff
@@ -153,7 +153,11 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 		return "templaterouter", nil
 	}
 
-	numSeconds := int(cfg.reloadInterval.Seconds())
+	numSeconds := int(router.reloadInterval.Seconds())
+	if numSeconds < 0 {
+		numSeconds = 5
+		glog.Errorf("Router --interval set < 0. Using 5 second reload interval")
```
Comment (deads2k) [inline pkg/router/template/router.go]: If I've rejected an explicit setting, I think I'd prefer to fail them as opposed to simply choosing a different value.

Is this for backwards compatibility?  If so, when would we tighten it?

Comment (eparis) [inline pkg/router/template/router.go]: New feature, no backwards compat requirements. I can glog.fatal it.

**Comment by eparis** [comment]
@deads2k better?

**Comment by deads2k** [comment]
> @deads2k better?

I guess I'm not familiar with the code, but given the method signature I expected `return nil, fmt.Errorf("message")`, but I do like failing better than ignoring.

**Pull request closed without merging**